src/main/java/reactor/core/publisher/FluxOnBackpressureBuffer.java

```diff
@@ -119,6 +114,29 @@ public long getPrefetch() {
 		}
 
 		@Override
+		public Object scan(Attr key) {
+			switch (key) {
+				case PARENT:
+					return s;
+				case REQUESTED_FROM_DOWNSTREAM:
+					return requested;
+				case TERMINATED:
+					return done && queue.isEmpty();
+				case CANCELLED:
+					return s == Operators.cancelledSubscription();
```

Comment:
wrong test, should be `return cancelled;` as cancel() doesn't set to cancelledSubscription